workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/deploy/HyscaleDeployServiceCommand.java

```diff
@@ -112,44 +131,51 @@
     private DeployComponentInvoker deployComponentInvoker;
     
     @Autowired
-    private ServiceSpecMapper serviceSpecMapper;
-
+    private ManifestAndDeployHelper manifestAndDeployHelper;
+    
     @Override
     public Integer call() throws Exception {
         if (!CommandUtil.isInputValid(this)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
+        HyscaleCommandSpec commandSpec = new HyscaleCommandSpec();
+        commandSpec.setAppName(appName);
+        commandSpec.setServiceSpecFiles(serviceSpecs);
+        if (profileInput != null) {
+            commandSpec.setProfileFiles(profileInput.getProfiles());
+            commandSpec.setProfileName(profileInput.getProfileName());
+        }
+        
+        if (!inputSpecValidator.validate(commandSpec)) {
+            return ToolConstants.INVALID_INPUT_ERROR_CODE;
+        }
+        List<File> profiles = commandSpec.getProfileFiles();
         
-        Map<String, File> serviceProfileMap = new HashMap<String, File>();
-        try {
-            serviceProfileMap = ServiceProfileUtil.getServiceProfileMap(profiles);
-        } catch (HyscaleException e) {
-            WorkflowLogger.error(ControllerActivity.CANNOT_PROCESS_SERVICE_PROFILE, e.getMessage());
-            throw e;
+        List<EffectiveServiceSpec> effectiveServiceSpecList = serviceSpecProcessor.getEffectiveServiceSpec(serviceSpecs,
+                profiles);
+        
+        List<WorkflowContext> contextList = manifestAndDeployHelper.getContextList(effectiveServiceSpecList, appName, null);
+        
+        manifestAndDeployHelper.getDeployPostValidators().forEach( each -> inputSpecPostValidator.addValidator(each));    
+        if (!inputSpecPostValidator.validate(contextList)) {
+            WorkflowLogger.logPersistedActivities();
+            return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
         
         boolean isCommandFailed = false;
-        for (File serviceSpecFile : serviceSpecs) {
+        for (EffectiveServiceSpec effectiveServiceSpec : effectiveServiceSpecList) {
             boolean isServiceFailed = false;
```

Comment:
Better move this isServiceFailed Logic to invoker caller itself

Reply:
We can discuss on this.